src/homepage/howTo/deleted-messages-vault.html

```diff
@@ -0,0 +1,322 @@
+                        </li>
+                    </ul>
+
+                    <header class="major">
+                        <h3><b>Make James uses Deleted Messages Vault by configuring Pre Deletion Hooks</b></h3>
+                    </header>
+
+                    <p>
+                        By default, although Deleted Messages Vault has been configured, but, to make it really work, you still need to configure Pre Deletion Hooks to lets James use it.
+                        Before deleting a mail in James, PreDeletionHooks will be triggered to execute all hooks, if all hook executions success, then Jame will process to delete that mail.
+                        There is already a DeletedMessageVaultHook in James, it's job is storing deleted mails into Deleted Messages Vault. Thus, you need to configure this hook in listener.xml configuration file.
+                    </p>
+
+                    <p>
+                        Sample DeletedMessageVaultHook configuration:
+                    </p>
+                    <pre><code>
+    &lt;listeners&gt;
+        &lt;listener&gt;
+        ...
+        &lt;/listener&gt;
+        ...
+        &lt;preDeletionHook&gt;
+            &lt;class&gt;org.apache.james.vault.DeletedMessageVaultHook&lt;/class&gt;
+        &lt;/preDeletionHook&gt;
+    &lt;/listeners&gt;
+                    </code></pre>
+
+                    <header class="major">
+                        <h3><b>Starting James with enabled Deleted Message Vault by docker compose</b></h3>
+                    </header>
+
+                    <p>We will take James cassandra product for the example</p>
+
+                    <p>First, get template cassandra product configuration:</p>
+                    <pre><code>
+    $ git clone https://github.com/apache/james-project
+    $ cp -rf james-project/dockerfiles/run/guice/cassandra/destination/conf conf
+    $ mv keystore conf/keystore
+                    </code></pre>
+
+                    <p>Second, modify deletedMessageVault.properties configuration file</p>
+                    <pre><code>
+    urlPrefix=cassandra://var/deletedMessages/vault
+    retentionPeriod=1 year
+                    </code></pre>
+
+                    <p>Third, modify listener.xml to configure DeletedMessageVaultHook by adding preDeletionHook section at above paragraph</p>
+
+                    <p>Fourth, We will create a local folder for holding data out of the container:</p>
+                    <pre><code>
+    mkdir var
+                    </code></pre>
+
+                    <p>Finally, starting a James Server by docker compose</p>
+                    <pre><code>
+    version: '2'
+
+    services:
+      james:
+        image: linagora/james-cassandra-guice:latest
+        ports:
+          - "80:80"
+          - "25:25"
+          - "110:110"
+          - "143:143"
+          - "465:465"
+          - "587:587"
+          - "993:993"
+          - "8000:8000"
+        volumes:
```

Comment:
I don't like copying the whol docker compose here.

Can't we say:

 - dowload the docker-compose
 - Adds the following volumes for `james` service:

```
    volumes:
          - $PWD/conf:/root/conf/
          - $PWD/var:/root/var/
```